Examples/tracking-pointer.lisp

```diff
@@ -0,0 +1,151 @@
+    (setf (medium-transformation (sheet-medium pane))
+          (make-scaling-transformation* 2 2))
+    (format pane "This pane has a scaling medium-transformation (2x2).~%"))
+  (with-output-as-presentation (pane 42 'integer)
+    (format pane "Presentation integer 42."))
+  (fresh-line pane)
+  (with-output-as-presentation (pane "42" 'string)
+    (format pane "Presentation string \"42\"."))
+  (fresh-line pane)
+  (with-output-as-presentation (pane "42" 'foo)
+    (format pane "Presentation FOO \"42\"."))
+  (fresh-line pane)
+  (with-room-for-graphics (pane)
+    (draw-rectangle* pane 10 10 50 50 :ink +dark-blue+)))
+
+(let ((last-sheet nil)
+      (last-region nil))
+  (defun draw-cursor (window x y id)
+    (when (and (rl *application-frame*) last-sheet)
+      (handle-repaint last-sheet (or last-region +everywhere+)))
+    (if (output-recording-stream-p window)
+        (with-output-recording-options (window :record nil)
+          (draw-point* window x y :line-thickness 32)
+          (draw-point* window x y :line-thickness 24 :ink +red+))
+        (progn
+          (draw-point* window x y :line-thickness 32)
+          (draw-point* window x y :line-thickness 24 :ink +red+)))
+    (setf last-sheet window
+          last-region (make-rectangle* (- x 16) (- y 16)
+                                       (+ x 16) (+ y 16)))
+    (window-clear *pointer-documentation-output*)
+    (format *pointer-documentation-output* "Press SPACE to exit.~%")
+    (format *pointer-documentation-output* "Window: ~4a, Clause ID: ~s.~%" (pane-name window) id)
+    (finish-output *pointer-documentation-output*)))
+
+(define-tracking-pointer-test-command (track-pointer :name t) ()
+  (let* ((tracked-sheet (find-pane-named *application-frame* (ts *application-frame*))))
+    (tracking-pointer (tracked-sheet :multiple-window (mwp *application-frame*)
+                                     :transformp (tp *application-frame*)
+                                     :highlight t
+                                     :context-type (ct *application-frame*))
+      (:pointer-motion
+       (&key window x y)
+       (draw-cursor window x y :pointer-motion))
+      (:pointer-button-press
+       (&key event x y)
+       (draw-cursor (event-sheet event) x y :pointer-button-press))
+      (:pointer-button-release
+       (&key event x y)
+       (draw-cursor (event-sheet event) x y :pointer-button-release))
+      (:presentation
+       (&key presentation window x y)
+       (declare (ignore presentation))
+       (draw-cursor window x y :presentation))
+      (:presentation-button-press
+       (&key presentation event x y)
+       (declare (ignore presentation event x y)))
+      (:presentation-button-release
+       (&key presentation event x y)
+       (declare (ignore presentation event x y)))
+      (:keyboard
+       (&key gesture)
+       ;; when gesture is a space then it should be already character.
+       (when (eventp gesture)
```

Comment:
because mcclim does not do what "should be" done. I'm not 100% certain about that either: should we call process-gesture with sheet as an argument? or simply take the character? or should we call stream-read-gesture when :multiple-window is nil?

Reply:
I have no idea. We can leave that for some other time.